plugins/destination/postgresql/client/migrate.go

```diff
@@ -48,165 +167,193 @@ func (c *Client) Migrate(ctx context.Context, tables schema.Tables) error {
-			if c.enabledPks() && col.CreationOptions.PrimaryKey {
-				reCreatePrimaryKeys = true
-			}
-			sql := "alter table " + tableName + " drop column " + columnName
-			// right now we will drop the column and re-create. in the future we will have an option to automigrate
-			if _, err := c.conn.Exec(ctx, sql); err != nil {
-				return fmt.Errorf("failed to drop column %s on table %s: %w", col.Name, table.Name, err)
-			}
-			sql = "alter table " + tableName + " add column " + columnName + " " + columnType
-			if _, err := c.conn.Exec(ctx, sql); err != nil {
-				return fmt.Errorf("failed to add column %s on table %s: %w", col.Name, table.Name, err)
-			}
-		}
+func (c *Client) alterPKConstraint(ctx context.Context, pgTable *schema.Table, table *schema.Table) error {
+	c.logger.Info().Str("table", table.Name).Msg("Recreating primary keys")
+	tableName := pgx.Identifier{table.Name}.Sanitize()
 
-		// column exists and type is the same but constraints might differ
-		if c.enabledPks() && pgPKs[col.Name] != col.CreationOptions.PrimaryKey {
-			c.logger.Info().Str("table", table.Name).Str("column", col.Name).Bool("pk", col.CreationOptions.PrimaryKey).Msg("Column exists with different primary keys")
-			reCreatePrimaryKeys = true
+	tx, err := c.conn.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction to recreate primary keys: %w", err)
+	}
+	dropConstraintSQL := "alter table " + tableName + " drop constraint if exists " + pgTable.PkConstraintName
+	if _, err := tx.Exec(ctx, dropConstraintSQL); err != nil {
+		if err := tx.Rollback(ctx); err != nil {
+			c.logger.Error().Err(err).Msg("failed to rollback transaction")
 		}
+		return fmt.Errorf("failed to drop primary key constraint on table %s: %w", table.Name, err)
 	}
 
-	if reCreatePrimaryKeys {
-		c.logger.Info().Str("table", table.Name).Msg("Recreating primary keys")
-		if err := c.setNotNullOnPks(ctx, table); err != nil {
-			return fmt.Errorf("failed to enforce not null on primary keys: %w", err)
+	sql := "alter table " + tableName + " add constraint primary key (" + strings.Join(table.PrimaryKeys(), ",") + ")"
+	if _, err := tx.Exec(ctx, sql); err != nil {
+		if err := tx.Rollback(ctx); err != nil {
+			c.logger.Error().Err(err).Msg("failed to rollback transaction")
 		}
+		return fmt.Errorf("failed to add primary key constraint on table %s: %w", table.Name, err)
+	}
+	if err := tx.Commit(ctx); err != nil {
+		return fmt.Errorf("failed to commit transaction to recreate primary keys: %w", err)
+	}
+	return nil
+}
 
-		tx, err := c.conn.Begin(ctx)
-		if err != nil {
-			return fmt.Errorf("failed to begin transaction to recreate primary keys: %w", err)
+func (c *Client) autoMigrateTable(ctx context.Context, pgTable *schema.Table, table *schema.Table) error {
+	changedColumns, changedPgColumns := table.GetChangedColumns(pgTable)
+	if c.spec.MigrateMode != specs.MigrateModeForced {
+		if changedColumns != nil {
```

Comment:
Non blocking but I think we can change the order of the errors reported here (first report PK error), since if we add a PK to an existing column users will see:
```
postgres table aws_apigateway_domain_name_base_path_mappings has different types 
for columns [account_id:TypeString, arn:TypeString] but schema wants 
[account_id:TypeString:PK:NotNull, arn:TypeString:PK:NotNull] , 
add `migrate_mode: forced` to the destination spec to drop the column
```

So a PK addition to an exsiting column is reported as a column change (due to the `NotNull` addition).
If we change the order of how we report errors I think it will be more clear what what the cause of the failure, WDYT?

Reply:
I think it should be in this order because in postgres you first need to make sure there is `notnull` on the column before you add a pk and it can't be in the same transaction.

Reply:
Ah, I think I didn't explain correctly. By different order I mean:
```go
		if c.enabledPks() && !table.IsPrimaryKeyEqual(pgTable) {
			return fmt.Errorf("postgres table %s has different primary keys %v but schema wants %v , add `migrate_mode: forced` to the destination spec to drop the column", table.Name, pgTable.PrimaryKeys(), table.PrimaryKeys())
		}
		if changedColumns != nil {
			return fmt.Errorf("postgres table %s has different types for columns %v but schema wants %v , add `migrate_mode: forced` to the destination spec to drop the column", table.Name, changedPgColumns, changedColumns)
		}
```

Instead of the current code, so the more specific error is printed first

Reply:
I see. reordered.